src/runtime/str.cpp

```diff
@@ -46,16 +46,33 @@ extern "C" PyObject* string_splitlines(PyStringObject* self, PyObject* args) noe
 
 namespace pyston {
 
-BoxedString::BoxedString(const char* s, size_t n) : s(s, n) {
-    gc::registerGCManagedBytes(this->s.size());
+BoxedString::BoxedString(const char* s, size_t n) : s(storage(), n) {
+    if (s) {
+        memmove(storage(), s, n);
+        storage()[n] = 0;
+    } else {
+        memset(storage(), 0, n + 1);
+    }
+    gc::registerGCManagedBytes(cls->tp_basicsize + this->s.size() + 1);
```

Comment:
I don't think we have to call this any more; it was there since the gc wouldn't see the allocation that the s constructor (as a std::string) would do.  I think this should get seen now by the allocation for the object in operator new?